postgres_storage.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"fmt"
+	"github.com/jackc/pgconn"
+	_ "github.com/jackc/pgx/v4/stdlib"
+	"github.com/jmoiron/sqlx"
+	uuid "github.com/satori/go.uuid"
+	"strings"
+)
+
+type PostgresStorage struct {
+	db *sqlx.DB
+}
+
+const connectionString = "postgres://postgres:postgres@localhost:5432/wasfaty?sslmode=disable"
+
+func NewPostgresStorage() (*PostgresStorage, error) {
+	db, err := sqlx.Connect("pgx", connectionString)
+	if err != nil {
+		return nil, err
+	}
+	return &PostgresStorage{db}, nil
+}
+
+func (s *PostgresStorage) GetAll() []*Person {
+	var pp []*Person
+
+	err := s.db.Select(&pp, `SELECT * FROM person`)
+	if err != nil {
+		return nil
+	}
+
+	for _, p := range pp {
+		var cc []*Communication
+		err = s.db.Select(&cc, fmt.Sprintf(`SELECT Value
+			FROM Communication
+			WHERE PersonId = '%s'`, p.ID.String()))
+		if err == nil {
+			p.Communications = cc
+		}
+	}
+
+	return pp
+}
+
+func (s *PostgresStorage) Add(p *Person) error {
+	_, err := s.db.Exec(fmt.Sprintf(`INSERT INTO person (id, name) VALUES ('%s', '%s')`, p.ID.String(), p.Name))
+	if err != nil {
+		if pgerr, ok := err.(*pgconn.PgError); ok {
+			if pgerr.Code == "23505" {
+				return personExistError
+			}
+		}
+		return err
+	}
+
+	for _, com := range p.Communications {
+		_, err = s.db.Exec(fmt.Sprintf(`INSERT INTO communication (value, personid) VALUES ('%s', '%s')`, com.Value, p.ID.String()))
+	}
+
+	return err
+}
+
+func (s *PostgresStorage) GetPersonById(id uuid.UUID) *Person {
+	var pp []*Person
+	err := s.db.Select(&pp, fmt.Sprintf(`SELECT * FROM person WHERE id = '%s' LIMIT 1`, id.String()))
```

Comment:
id is primary key, no limit 1 is needed
use get row instead of select https://jmoiron.github.io/sqlx/#queryrow
but don't forget to handle ErrNoRows